### PR TITLE
Fix algorithm version constant MIXED_REFERENCE_MODE for DDR

### LINKED ISSUE #11801: Mixed builds, gccheck test failing, DDR problem?
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_s390x_linux_mixed_Nightly_testList_0/1/
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_s390x_linux_mixed_Nightly_testList_1/1/
cmdLineTester_GCCheck (all modes)
```
 [OUT] Checking HEAP...  <gc check (1): from debugger: HEAP: Object 3ff0d590008: corrupt data exception>
 [OUT]   <gc check (1): Corrupt 3ff0d590008 header: 00000000 00000000>
 [OUT]   <gc check (2): from debugger: HEAP: Object 3ff88400008: corrupt data exception>
 [OUT]   <gc check (2): Corrupt 3ff88400008 header: 00000000 00000000>
 [OUT]   <gc check (3): from debugger: HEAP: Object 3ff88500000: corrupt data exception>
 [OUT]   <gc check (3): Corrupt 3ff88500000 header: 00000000 00000001>
 [OUT] done (39 ms).
 [OUT] Checking CLASS HEAP...  <gc check (4): from debugger: CLASS HEAP: slots slot 3ff903b2d00(3ff903b2d30) -> 3ff0d5ec5b0: class pointer not in a class segment>
 [OUT]   <gc check (5): from debugger: CLASS HEAP: static slot 3ff903b2900(3ff903b2858) -> 3ff884a7500: class pointer not in a class segment>
 [OUT]   <gc check (6): from debugger: CLASS HEAP: slots slot 3ff903b2300(3ff903b2330) -> 3ff0d5ec2c0: class pointer not in a class segment>
 [OUT]   <gc check (7): from debugger: CLASS HEAP: static slot 3ff903b2000(3ff903b21b8) -> 3ff884a74d8: class pointer not in a class segment>
 [OUT]   <gc check (8): from debugger: CLASS HEAP: static slot 3ff903b1e00(3ff903a5178) -> 3ff884a74c0: class pointer not in a class segment>
...
```

This is a newly ported test, in case you are wondering why we didn't see this fail previously.

@sharon-wang @dmitripivkine @keithc-ca fyi

### PR DESCRIPTION
**Pull request by keithc-ca**
Fixes #11801.

### CODE REVIEW

**Comment by gacholio** [comment]
This seems like a non-change - how is this fixing anything?

**Comment by keithc-ca** [comment]
The constant `MIXED_REFERENCE_MODE` is only visible to the new DDR tooling if it is a macro. The table `AlgorithmVersions` is only used by the `j9ddrgen` executable which is only built and used by internal builds.

**Review by gacholio — APPROVED** [review]

File: runtime/ddr/algorithm_versions.c [hunk 1]
```diff
@@ -63,9 +70,9 @@ J9DDRConstantTableBegin(DDRAlgorithmVersions)
 	J9DDRConstantTableEntryWithValue("ALG_VM_ITABLE_VERSION", ALG_VM_ITABLE_VERSION)
 	J9DDRConstantTableEntryWithValue("ALG_VM_BYTECODE_VERSION", ALG_VM_BYTECODE_VERSION)
 	J9DDRConstantTableEntryWithValue("ALG_OBJECT_MONITOR_VERSION", ALG_OBJECT_MONITOR_VERSION)
-#if defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS)
-	J9DDRConstantTableEntryWithValue("MIXED_REFERENCE_MODE", 1)
-#endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
+#ifdef MIXED_REFERENCE_MODE
```
Comment (gacholio) [inline runtime/ddr/algorithm_versions.c:73]: We always prefer if defined.

Comment (keithc-ca) [reply runtime/ddr/algorithm_versions.c]: Fixed.

**Comment by gacholio** [comment]
Given how trivial this change is, I'm not going to consume build resources for it.

**Pull request merged**